src/Storage/Doctrine/Connection.php

```diff
@@ -98,6 +102,39 @@ public function findMessage(string $id): ?StoredMessage
         );
     }
 
+    public function getStatistics(\DateTimeImmutable $from, \DateTimeImmutable $to): Statistics
+    {
+        $statement = $this->executeQuery(
+            $this->driverConnection->createQueryBuilder()
+                ->select('count(id) as countMessagesOnPeriod, class')
+                ->addSelect(sprintf('AVG(%s) AS averageWaitingTime', $this->SQLDriver->getDateDiffInSecondsExpression('received_at', 'dispatched_at')))
+                ->addSelect(sprintf('AVG(%s) AS averageHandlingTime', $this->SQLDriver->getDateDiffInSecondsExpression('handled_at', 'received_at')))
+                ->from($this->tableName)
+                ->where('handled_at >= :from')
+                ->andWhere('handled_at <= :to')
+                ->groupBy('class')
+                ->getSQL(),
+            ['from' => $from, 'to' => $to],
+            ['from' => Types::DATETIME_IMMUTABLE, 'to' => Types::DATETIME_IMMUTABLE,]
+        );
+
+        $statistics = new Statistics($from, $to);
+        while (false !== ($row = $statement->fetch(FetchMode::ASSOCIATIVE))) {
+            $statistics->add(
+                new MetricsPerMessageType(
+                    $from,
+                    $to,
+                    $row['class'],
+                    (int) $row['countMessagesOnPeriod'],
+                    (float) $row['averageWaitingTime'],
+                    (float) $row['averageHandlingTime']
+                )
+            );
+        }
+
+        return $statistics;
+    }
+
```

Comment:
It occurred to me that, even if we don't use it yet, we should probably store the string `receiverName` on the database table / persistence